### DISCUSSION TITLE
Campaign AI unfair handling of starting units

### DISCUSSION THREAD

**Discussion by xhrana02** [ihhub/fheroes2#6981] [General]
I'm going through the original evil campaign and some missions seem just ridiculously hard. The worst offender so far is the optional mission 7 (Dragon Island or something) which seems just hopelessly unwinnable.

And I think I realized why. I'm basing this on vague memories from 15 years ago so someone correct me if I'm wrong. I vaguely remember that in the original game the AI had in some missions pretty strong starting armies but they were forced to keep them in their castle as a sort of "final boss" guard.

But now they just take it all with them and wipe the whole map within the first few weeks.

I'm also playing the campaign on the Hard difficulty so I don't know if it's caused by the difficulty setting or by fheroes2 breaking something in the campaign AI but I would bet it on the broken AI.

**Comment by ihhub** [comment]
Hi @xhrana02 , this topic was discussed few times in the past. The original AI wasn't forced to keep armies in castles, it was over defensive in general. Therefore, it keeps all troops there. The original AI is dumb in many ways and has a lot of flaws which players can abuse.  These flaws were not introduced in fheroes2 AI as they are absolutely different instances. fheroes2 AI is written from scratch. Hence, AI works perfectly fine as it should be, the logic is very close to a player game style. In the situation of this particular campaign AI is smart enough to use a chunky startup army bonus and give armies to the boldest hero, which is a very correct decision if it were a human player. So saying it as a broken AI is not a right thing. Nothing wrong with this behavior.

On the other hand, the map was designed in a way to compensate dumb AI by giving it more armies and resources. The original AI has huge extra bonuses to army growth and to accumulated resources. fheroes2 has slightly reduced bonuses in comparison and going to have even smaller since AI is becoming smart enough.

There are multiple ways to beat this scenario which was discussed either here or in Discord. Please refer to them as a starting point.

**Comment by ihhub** [comment]
Also playing on Hard difficulty means that AI has bigger bonuses to army growth and resources so it can expand the occupied territory faster.

**Comment by xhrana02** [comment]
Ok, thx for the reply.
I personally think that the original AI suits the campaigns better. I don't find this agressive human-like AI enjoyable in the campaigns. You're supposed to slowly go through the content of the entire map to familiarize yourself with all the creatures and interactable objects in the game and then siege the AI castles (the "final bosses"). The fheroes2 AI basically denies you the opportunity to interact with most of the map's contents. Campaigns aren't suppossed to feel like unfair PvP where you have to savescum, minmax, and exploit AI shortcomings in order to have at least some chance of winning.

Also I think that any army growth / resources bonuses for the AI are irrelevant when the problem is their starting armies. Although somewhere I read that the difficulty affects the player's starting resources too, which could have a pretty big impact on the balance.

As for how to beat this specific mission - I've searched the discussions here both before and after posting and I've found nothing (I didn't check the Discord server yet). Could you please point me in the right way?

**Reply by zenseii** [reply]
Hi, @xhrana02 ! I'm happy to hear you checking out the project!

We've discussed this exact mission which having the Ogre Alliance bonus can mean the difference between a walk in the park or not.

I have referenced a youtube video of a skilled player showing how it can be beat without the alliance.

Also about your point about how the campaigns were made to let a player familiarize themselves with the game. We have had discussions about this with long time players and many claim the opposite - that the original campaigns were something you should only try once you've played the included tutorial* and several scenarios.

Similarly, the expansion campaigns were originally very hard. I know many longtime players who had to quit them when playing the original game with the original AI. They are harder with the fheroes2 AI when playing on hard, as it should be, but they are definitely not unbeatable as I regularly see players beat them.

I will try to find this video once I have time to. It is by someone called Nimethua games.

*The tutorial map that comes with the game is incompatible with fheroes2 since it is in the form of a save file, and fheroes2 has its own save file system. Someone remade that map as a standard map file but due to Ubisoft's license of the game we aren't allowed to even share user-made map files. Yes, this is weird and has never been a problem, but we've decided to stay on the safe side and respect it.

**Reply by xhrana02** [reply]
Hi, @zenseii 
I've checked Nimethua games YT channel and I found the specific video https://www.youtube.com/watch?v=eYLL9DtmSnE
But the AI in his game behaves a bit differently from the AI in my game. In my game the Orange player consistently brings his strongest hero with all of his dragons and minotaurs into the middle of the map and clears it out around turn 4 I think. If I don't rush there and clear the trail of resources, the AI will follow the resources and then attack my castle during the first week. If I do rush it a clear them first, the AI then circles around for another week before getting to my castle. The yellow player brings his dragons around the same time.

But in Nimethua's game the Orange player doesn't bother clearing the middle at all. And overall it feels far less agressive (although this part may be due to him spreading quite quickly to their castles and the AI going on the defensive).
I saw some ways in which to improve my strategy in the video so maybe it will be enough to help me win the mission.

**Reply by zenseii** [reply]
Yes, Nimethua games played using version 1.0 AI and we have already made improvements since then. 

Please keep us updated on how it goes. I might try my hand at it at some point.

**Reply by xhrana02** [reply]
Nah... I've lost count of the attempts, even copying his strategy step by step, not even close. In his video the AI seems to wander with their main forces somewhere in the deserts in the corner of the maps so Nimethua is free to completely control the center and do whatever he wants without ever having to worry about defense. The AI doesn't even really bother to defend their castles that much.

In my attempts, the AI is controlling the center from the start and just patrols around their castles and around the center making it extremely difficult to do anything. And the player's castle is sooo close to the center that 1 misstep means instant defeat.

**Reply by xhrana02** [reply]
I'll keep trying though.

**Reply by ihhub** [reply]
@xhrana02 , are you using latest release or snapshot of fheroes2? Recently AI got an upgrade for fog discovery.

**Reply by xhrana02** [reply]
@ihhub latest release (1.0.2)... if that's without the fog discovery upgrade then mercy on the poor souls trying this mission in the future.

**Reply by xhrana02** [reply]
@zenseii So I've finally beaten the mission. It took me basically the whole of today + a several hours from previous days. I had to savescum almost every turn. Math didn't even invent the number describing the amount of resets I had to do. And I had to exploit the AI pretty heavily. Mainly I had to have a few heroes full-time dedicated to running around the enemy castles to pull the agro of their main heroes. And all fights had to revolve around vampire lords and never letting any of them die, the rest of the units was basically cannon fodder for enemy spells. I don't even want to imagine how someone would want to beat the map using Warlock / Barbarian (without Ogre Alliance).

I also got lucky cause one of the AIs managed to lose one of their starting dragons against neutrals. Then they lost another one to their own turrets when recapturing their castle. And with only 1 dragon remaining it was pretty easy to finish that AI. Then it was just waiting for the superior income from 2 castles vs 1 to finish the job.

But jeez... this took me more time than all the previous missions combined, several times over!

It might've been faster to just replay the entire campaign from the start and take the Ogre Alliance bonus instead of the Necro guy.
